apps/balancer-tools/src/app/apr/round/[roundId]/page.tsx

```diff
@@ -1,13 +1,58 @@
 "use client";
 
 import { useParams } from "next/navigation";
+import { useEffect, useState } from "react";
+
+import Loading from "#/app/metadata/[network]/pool/[poolId]/loading";
+import { Row } from "#/lib/dune";
+
+import PoolCard from "../(components)/PoolCard";
 
 export default function Page() {
+  const [poolsData, setPoolsData] = useState<Row[]>([]);
+  const [hasError, setHasError] = useState("");
   const { roundId } = useParams();
+  useEffect(() => {
+    if (!poolsData.length) {
+      const fetchData = async () => {
+        const poolsDataRows = await (
```

Comment:
This made me realise maybe this component shouldnt be client

Reply:
I think there's nothing wrong in doing it this way -- but think we'll end up needing to write less code if we do the server components approach. Also we'll learn something new 😉